source/Digitisers/src/DDPlanarDigiProcessor.cc

```diff
@@ -308,10 +348,41 @@ void DDPlanarDigiProcessor::processEvent( LCEvent * evt ) {
           continue; 
         }
       }
+
+      //***************************************************************
+      // Smear time of the hit and apply the time window cut if needed
+      //***************************************************************
       
-      //**************************************************************************
-      // Try to smear the hit but ensure the hit is inside the sensitive region
-      //**************************************************************************
+      // Smearing time of the hit
+      float resT = _resT.size() > 1 ? _resT.at(layer) : _resT.at(0); 
+      double tSmear  = resT == 0.0 ? 0.0 : gsl_ran_gaussian( _rng, resT );
+      _h[hT]->Fill( resT == 0.0 ? 0.0 : tSmear / resT );
+      _h[diffT]->Fill( tSmear );
+
+      // Skipping the hit if its time is outside the acceptance time window
+      double hitT = simTHit->getTime() + tSmear;
+      streamlog_out(DEBUG3) << "smeared hit at T: " << simTHit->getTime() << " ns to T: " << hitT << " ns according to resolution: " << resT << " ns" << std::endl;
+      
```

Comment:
```suggestion
      } // if time smearing enabled
```
As time correction or time window selection might be interesting also without smearing